pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1189,6 +1192,71 @@ func (c *Cluster) saveMonConfig() error {
 }
 
 func (c *Cluster) persistExpectedMonDaemons() error {
+	if err := c.persistExpectedMonDaemonsInConfigMap(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in ConfigMap")
+	}
+
+	if err := c.persistExpectedMonDaemonsAsEndpointSlice(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in EndpointSlice")
+	}
+	return nil
+}
+
+func (c *Cluster) persistExpectedMonDaemonsAsEndpointSlice() error {
+	monitors := c.ClusterInfo.AllMonitors()
+
+	if len(monitors) == 0 {
+		logger.Debug("no mon addresses found, skipping endpointslice resource creation")
+		return nil
+	}
+
+	monAddresses := make([]string, 0, len(monitors))
+
+	for _, mon := range monitors {
+		ip := strings.Split(mon.Endpoint, ":")[0]
+		monAddresses = append(monAddresses, ip)
+	}
+
+	endpointSlice := &discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      endpointSliceName,
+			Namespace: c.Namespace,
+			Labels: map[string]string{
+				"kubernetes.io/service-name": endpointSliceName,
+			},
+		},
+		Endpoints: []discoveryv1.Endpoint{
+			{
+				Addresses: monAddresses,
+			},
+		},
+		Ports:       nil,
+		AddressType: "IPv4",
```

Comment:
If the mon pod is running on IPV6 only will this also work without any issue?

Reply:
Can we generate the endpoints based on the [ipFamily and dualStack settings](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L130-L132)? No need to create the endpoint slice for a protocol that's not enabled, it should be a simple check of those settings.